### PR TITLE
WT-8660 Make bucket name and prefix as part of the S3Connection class

### PR DESCRIPTION
**Pull request by sulabhM**

### CODE REVIEW

File: test/suite/test_s3_store01.py [hunk 1]
```diff
@@ -68,8 +78,8 @@ def test_local_basic(self):
         os.remove(cache_prefix + bucket_name + '/' + filename)
         self.assertTrue(fs.fs_exist(session, filename))
 
-        fs2 = s3_store.ss_customize_file_system(session, "wt-bucket", "Secret", None)
-        _ = fs2.fs_directory_list(session, self.bucket_name, '')
+        fs2 = s3_store.ss_customize_file_system(session, self.bucket_name, "Secret", self.fs_config)
+        self.assertEquals(fs2.fs_directory_list(session, self.bucket_name, ''), [object_name, "blah"])
```
Comment (sulabhM) [inline test/suite/test_s3_store01.py:82]: The "blah" is temporary to test a failure! Will be removed from final change.

File: ext/storage_sources/s3_store/s3_storage_source.cpp [hunk 2]
```diff
@@ -238,49 +259,36 @@ S3CustomizeFileSystem(WT_STORAGE_SOURCE *storageSource, WT_SESSION *session, con
 
     /* TODO: Move these into tests. Just testing here temporarily to show all functions work. */
```
Comment (sulabhM) [inline ext/storage_sources/s3_store/s3_storage_source.cpp:260]: The final change for this PR will remove this test block - when all tests are green.

File: test/suite/test_s3_store01.py [hunk 3]
```diff
@@ -68,8 +78,8 @@ def test_local_basic(self):
         os.remove(cache_prefix + bucket_name + '/' + filename)
```
Comment (ruby-oujo) [inline test/suite/test_s3_store01.py:78]: self.bucket_name

Comment (ruby-oujo) [inline test/suite/test_s3_store01.py:78]: Missed this self.bucket_name 🙂

File: ext/storage_sources/s3_store/s3_log_system.cpp [hunk 4]
```diff
@@ -3,15 +3,14 @@
 #include <cstdarg>
 
 S3LogSystem::S3LogSystem(WT_EXTENSION_API *wtApi, uint32_t wtVerbosityLevel)
+    : _wtApi(wtApi), _wtVerbosityLevel(wtVerbosityLevel)
 {
     // If the verbosity level is out of range it will default to AWS SDK Error level.
     if (verbosityMapping.find(wtVerbosityLevel) != verbosityMapping.end()) {
-        awsLogLevel = verbosityMapping.at(wtVerbosityLevel);
+        _awsLogLevel = verbosityMapping.at(wtVerbosityLevel);
```
Comment (fionaochee) [inline ext/storage_sources/s3_store/s3_log_system.cpp:10]: Should braces be removed here?

File: ext/storage_sources/s3_store/s3_log_system.h [hunk 5]
```diff
@@ -6,14 +6,21 @@
...
+
 class S3LogSystem : public Aws::Utils::Logging::LogSystemInterface {
 
     public:
     S3LogSystem(WT_EXTENSION_API *wtApi, uint32_t wtVerbosityLevel);
     Aws::Utils::Logging::LogLevel
     GetLogLevel(void) const override
     {
-        return awsLogLevel;
+        return _awsLogLevel;
```
Comment (fionaochee) [inline ext/storage_sources/s3_store/s3_log_system.h:23]: Should the return be wrapped in brackets?
`return (_awsLogLevel);`

**Review by ruby-oujo — APPROVED** [review]
LGTM! 🙂

**Review by fionaochee — APPROVED** [review]
LGTM!

**Pull request merged**